src/tiff.ml

```diff
@@ -854,31 +872,46 @@ module Data = struct
 
   let ceil a b = (a + b - 1) / b
 
-  let read_data t ro window arr_type read_value =
+  let read_data t ro plane window arr_type read_value =
     let ifd = t.ifd in
-    let samples_per_pixel =
-      try Ifd.samples_per_pixel ifd with Not_found -> 1
-    in
+    let samples_per_pixel = Ifd.samples_per_pixel ifd in
     let height = Ifd.height ifd in
     let width = Ifd.width ifd in
     let bits_per_sample = Ifd.bits_per_sample ifd in
-    let bytes_per_pixel = List.fold_left Int.add 0 bits_per_sample / 8 in
+    let planar_configuration = Ifd.planar_configuration ifd in
+    let bytes_per_pixel =
+      match planar_configuration with
+      | Chunky -> List.fold_left Int.add 0 bits_per_sample / 8
+      | Planar -> List.nth bits_per_sample plane / 8
+      | _ ->
```

Comment:
I think it is best to make this `Unknown i` so that if we do at some point support other planar configurations the type checker will let us know this bit of code needs updated, otherwise it will silently break.

Reply:
I admire your optimism that the 1992 spec will be updated - but totally agree :)